packages/test/test-end-to-end-tests/src/test/gc/gcContainerRuntimeExpiry.spec.ts

```diff
@@ -47,22 +49,60 @@ import { TestDataObject } from "./mockSummarizerClient";
     );
 
     let container1: IContainer;
-
     const createContainer = async (): Promise<IContainer> => provider.createContainer(runtimeFactory);
+    const loadContainer = async (): Promise<IContainer> => provider.loadContainer(runtimeFactory);
+    let clock: SinonFakeTimers;
 
-    beforeEach(async () => {
-        provider = getTestObjectProvider();
+    before(() => {
+        clock = useFakeTimers();
     });
 
-    it("Container should be closed with a ClientSessionExpired error after the gcSessionExpiryTime is up", async () => {
+    afterEach(() => {
+        clock.reset();
+    });
+
+    after(() => {
+        clock.restore();
+    });
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider();
         container1 = await createContainer();
         container1.on("closed", (error) => {
             assert.strictEqual(error?.errorType, "clientSessionExpiredError");
         });
+    });
 
+    it("Container should be closed with a ClientSessionExpired error after the gcSessionExpiryTime is up", async () => {
         await provider.ensureSynchronized();
-        const delay = async (ms: number) => new Promise((res) => setTimeout(res, ms));
-        await delay(100);
-        assert(container1.closed === true, "Container should be closed");
+        clock.tick(timeoutMs - 1);
+        assert(container1.closed === false, "Container1 should not be closed, it should be 1 tick away from expiring.");
+        clock.tick(1);
+        assert(container1.closed, "Container1 should be closed, it has should have reached its session expiry timeout");
+    });
+
+    it("Containers should have the same expiry time for the same document", async () => {
+        // Container1 should expire in one tick
+        clock.tick(timeoutMs - 1);
+        // Load the two other containers
+        const container2 = await loadContainer();
+        const container3 = await loadContainer();
+        assert(container1.closed === false, "Container1 should not be closed, it should be 1 tick away from expiring.");
+        assert(container2.closed === false, "Container2 should not be closed, it should not expire on load.");
+        assert(container3.closed === false, "Container3 should not be closed, it should not expire on load.");
+        // Ticking one more should expire the first container
+        clock.tick(1);
+        assert(container1.closed, "Container1 should be closed as it has reached its expiry.");
+        // Containers 2 & 3 should be one tick in.
+        assert(container2.closed === false, "Container2 should not be closed, as it has not reached expiry.");
+        assert(container3.closed === false, "Container3 should not be closed, as it has not reached expiry.");
+        // Containers 2 & 3 should be one tick away from expiring
```

Comment:
This comment looks incorrect. They will be one tick away after the clock.tick below.